app/vmalert/group.go

```diff
@@ -163,6 +168,7 @@ func (g *Group) updateWith(newGroup *Group) error {
 	}
 	g.Type = newGroup.Type
 	g.Concurrency = newGroup.Concurrency
+	g.ExtraFilterLabels = newGroup.ExtraFilterLabels
```

Comment:
I'm not sure, that it will work. Because on `ExtraFilterLabels` its needed to rebuild querier. 

 Probably, its better to include it into ruleID calculation and recreate group.

Reply:
Looks like update logic becomes too complicated) I was confused at first as well. I think, I'll need to refactor it a bit.
The real change happens here https://github.com/VictoriaMetrics/VictoriaMetrics/pull/1319/files#diff-3f50ee1eaf18706cb49d30cd978eb5f7dcebfd339f08beb3125ac77a9672370fR254 - old rule inherits Querier from new rule.

Reply:
Get it! Thanks.